Dockerfile

```diff
@@ -14,13 +14,20 @@ WORKDIR ${APP_ROOT}
 RUN source ${APP_ROOT}/etc/scl_enable \
   && gem install bundler --version=2.0.1 --no-document
 
-COPY --chown=default:root Gemfile* ./
+USER root
+COPY Gemfile* ./
+RUN chown -fR default:root ./Gemfile
+
+USER default
 RUN source ${APP_ROOT}/etc/scl_enable \
   && bundle config build.pg --with-pg-config=/usr/pgsql-10/bin/pg_config \
   && bundle install --deployment --path vendor/bundle --jobs $(grep -c processor /proc/cpuinfo) --retry 3
 
-COPY --chown=default:root . .
+USER root
+COPY . .
+RUN chown -fR default:root .
 
```

Comment:
It would be nice to group them to this section https://github.com/3scale/zync/pull/331/files#diff-3254677a7917c6c01f55212f86c57fbfR17

Then having less layer

Reply:
> It would be nice to group them to this section https://github.com/3scale/zync/pull/331/files#diff-3254677a7917c6c01f55212f86c57fbfR17
> 
> Then having less layer

Yep. That's what I meant in the description with

> This significantly increases the size of the image built from 719MB to 846MB. We could mitigate this effect a little by changing the order of steps in the Dockerfile therefore creating less layers. The downside of it is that we might be running useless heavy steps such as `COPY . .` (followed by `chown`) in cases where `RUN bundle install --deployment` fails for example.

Nonetheless, we may end up closing this PR anyway.